src/mbgl/util/clip_lines.cpp

```diff
@@ -0,0 +1,66 @@
+#include "clip_lines.hpp"
+
+namespace mbgl {
+namespace util {
+
+std::vector<std::vector<Coordinate>> clipLines(const std::vector<std::vector<Coordinate>> &lines,
+        const int16_t x1, const int16_t y1, const int16_t x2, const int16_t y2) {
```

Comment:
The `box` (util/box.hpp) doesn't fit well. Is `Rect` (util/rect.hpp) what you were thinking of? or were you thinking of making a new type?